### PR TITLE
[REF] hr_recruitment: change override write for onchange compute

### PR DESCRIPTION
**Pull request by Julien00859**

### CODE REVIEW

**Comment by Julien00859** [comment]
@jbm-odoo :wave:

**Comment by jbm-odoo** [comment]
Looks good to me

File: addons/hr_recruitment/models/hr_recruitment.py [hunk 1]
```diff
@@ -129,9 +129,10 @@ def _default_company_id(self):
     categ_ids = fields.Many2many('hr.applicant.category', string="Tags")
     company_id = fields.Many2one('res.company', "Company", default=_default_company_id)
     user_id = fields.Many2one('res.users', "Responsible", tracking=True, default=lambda self: self.env.uid)
-    date_closed = fields.Datetime("Closed", readonly=True, index=True)
-    date_open = fields.Datetime("Assigned", readonly=True, index=True)
-    date_last_stage_update = fields.Datetime("Last Stage Update", index=True, default=fields.Datetime.now)
+    date_closed = fields.Datetime("Closed", readonly=True, index=True, compute='_compute_date_closed', store=True)
+    date_open = fields.Datetime("Assigned", readonly=True, index=True, compute='_compute_date_open', store=True)
+    date_last_stage_update = fields.Datetime("Last Stage Update", index=True, default=fields.Datetime.now,
+        compute='_compute_date_last_stage_update', store=True)
```
Comment (Julien00859) [inline addons/hr_recruitment/models/hr_recruitment.py]: * The default could be removed as this is automatically done by the computer
* The field can set as read-only as there is no reason for customer to manually edit that field

File: addons/hr_recruitment/models/hr_recruitment.py [hunk 2]
```diff
@@ -282,6 +283,27 @@ def _onchange_stage_id_internal(self, stage_id):
...
 
+    @api.depends('user_id')
+    def _compute_date_open(self):
+        if self.user_id:
+            self.write({'date_open': fields.Datetime.now()})
+
+    @api.depends('stage_id')
+    def _compute_kanban_state(self):
+        if self.stage_id:
+            self.write({'kanban_state': 'normal'})
```
Comment (Julien00859) [inline addons/hr_recruitment/models/hr_recruitment.py]: Before the kanban_state was reset to normal on stage_id update when no new kanban_state was supplied. It is now written every time.

**Comment by rco-odoo** [comment]
@robodoo r-

File: addons/hr_recruitment/models/hr_recruitment.py [hunk 3]
```diff
@@ -268,19 +269,24 @@ def onchange_partner_id(self):
...
+    @api.depends('user_id')
+    def _compute_date_open(self):
+        for rec in self:
+            if rec.user_id:
+                rec.date_open = fields.Datetime.now()
+
+    @api.depends('stage_id')
+    def _compute_kanban_state(self):
+        # when the stage changes, reset the kanban stage to normal
+        self.kanban_state = 'normal'
```
Comment (Julien00859) [inline addons/hr_recruitment/models/hr_recruitment.py]:
```suggestion
    test
    test
    test
    test
```

**Comment by Julien00859** [comment]
Il n'y a aucun intérêt buisness à ce que cette branche soit mergée en stable, mergons la en master, c'est bien. J'ai force push pour rebase.

**Pull request closed without merging**